app/Repositories/CacheUserRepository.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Gladiator\Repositories;
+
+use Gladiator\Models\User;
+use Illuminate\Support\Facades\Cache;
+
+class CacheUserRepository implements UserRepositoryInterface
+{
+    /**
+     * UserRepositoryInterface instance.
+     *
+     * @var \Gladiator\Repositories\UserRepositoryInterface
+     */
+    protected $database;
```

Comment:
Yeah, I was just afraid it wouldn't be as straightforward since you're _in_ the Cache Repository and then requesting from another repository might seem like requesting from the same one :confounded: I can update if it doesn't feel right though!